crates/cargo-platform/README.md

```diff
@@ -0,0 +1,2 @@
+> This crate is maintained by the Cargo team for use by the wider
+> ecosystem. This crate follows semver compatibility for its APIs.
```

Comment:
I'm on the fence here. My intention when I created it was to be more "internal". I was thinking that if others found it useful, we could expand that to a more "experimental" status. I didn't know if anyone would actually have a use for it, and it wasn't really something I wanted to put effort into. 

Also, as evident by the lack of a README, this isn't really up to my standards for an intentional artifact that we intend to support.

It looks like it has a few dependents now, though: https://crates.io/crates/cargo-platform/reverse_dependencies.

I'm fine with elevating this to be "intentional" if ya'll agree. It has required almost no maintenance effort, and I don't expect us to delete it, and we already have semver checks. I just wanted to make clear what I was originally thinking.

(We probably should eventually add a real README, though 🤣)

Reply:
Thanks for the clarification!

I think its reasonable for the Rust Project to provide an API for this micro-DSL.  My main concern is what shape any of that should take and thats where I feel like the commitment of "Intentional" is strong.  Granted, the definition of "Intentional" is strong enough I'm hesitant to mark *anything* intentional but anything else would also be too weak / scare people away.